bundle/Command/ImportTagsCommand.php

```diff
@@ -0,0 +1,165 @@
+        }
+
+        $parentTagId = $input->getOption('parent-tag-id');
+
+        if (empty($parentTagId)) {
+            $parentTagId = 0;
+        }
+
+        $fileFullPath = $this->getContainer()->getParameter('kernel.root_dir') . $filename;
+
+        if (!file_exists($fileFullPath)) {
+            throw new \RuntimeException("No file has been found on the specified location.");
+        }
+
+        $output->writeln('Found file on the specified location.');
+
+        $fileContents = file_get_contents($fileFullPath);
+
+        /** @var \Symfony\Component\Serializer\Serializer $serializer */
+        $serializer = $this->getContainer()->get('serializer');
+
+        $data = $serializer->decode($fileContents, 'csv');
+
+        $dataCount = count($data);
+
+        $output->writeln("Found <comment>{$dataCount}</comment> tags for import from the file.");
+
+        // We support only locale headers and the RemoteID header for now
+        $headers = array_keys($data[0]);
+
+        $localeList = array();
+
+        $output->writeln('Extracting the locale list.');
+
+        foreach ($headers as $header) {
+            if ($header !== 'RemoteID') {
+                $localeList[] = $header;
+            }
+        }
+
+        $output->writeln('List of locales extracted.');
+
+        // IMPORTANT: the mainLanguageCode for the tag is always presumed to be the first language in the list.
+        $mainLanguageCode = $localeList[0];
+
+        $existing = array();
+
+        $progress = new ProgressBar($output, $dataCount);
+
+        $progress->start();
+
+        foreach ($data as $datum) {
+            if (in_array('RemoteID', $headers)) {
+                $remoteID = $datum['RemoteID'];
+
+                try {
+                    /** @var \Netgen\TagsBundle\API\Repository\Values\Tags\Tag $tag */
+                    $tag = $tagsService->loadTagByRemoteId($remoteID);
+                } catch (\eZ\Publish\API\Repository\Exceptions\NotFoundException $e) {
+                    $this->createTag($datum, $mainLanguageCode, $localeList, $parentTagId);
+                }
+
+                if (isset($tag) && ($tag instanceof Netgen\TagsBundle\API\Repository\Values\Tags\Tag))
+                {
+                    $existing[] = $tag->remoteId;
+                    $progress->advance();
+                    continue;
+                }
+
+                $this->createTag($datum, $mainLanguageCode, $localeList, $parentTagId);
+                $progress->advance();
+            }
+        }
+
+        $progress->finish();
+
+        $output->writeln('');
+
+        $output->writeln("The import is complete.");
+
+        if (count($existing) > 0) {
+            $existingCount = count($existing);
+
+            $existingIds = implode(', ', $existing);
+
+            $output->writeln("<comment>{$existingCount}</comment> tags already exist. The remote IDs in question are: {$existingIds}");
```

Comment:
Why not output this as you import and come to a tag that exists instead of one big line of output? What if all tags exist?